controllers/machinehealthcheck_targets.go

```diff
@@ -185,11 +185,6 @@ func (r *MachineHealthCheckReconciler) getTargetsFromMHC(ctx context.Context, lo
 		return nil, nil
 	}
 
-	var cluster clusterv1.Cluster
-	if err := clusterClient.Get(ctx, client.ObjectKey{Namespace: mhc.Namespace, Name: mhc.Spec.ClusterName}, &cluster); err != nil {
-		return nil, errors.Wrapf(err, "error getting Cluster %s/%s for MachineHealthCheck %s", mhc.Namespace, mhc.Spec.ClusterName, mhc.Name)
-	}
```

Comment:
Shouldn't this client be `r.Client` and not the target Cluster client?

Reply:
Regardless of this, +1 passing the object in if we already have it

Reply:
Not sure if I got it correctly:
* the issue was that we were trying to get the cluster from the target cluster instead of the management cluster
* but we still need the clusterClient (as target cluster client) in l.211